app/controllers/concerns/user_invitation.rb

```diff
@@ -62,6 +72,22 @@ def invite_new_user(email:, login: nil, first_name: nil, last_name: nil)
   end
 
   ##
+  # Sends a new invitation to the user with a new token.
+  #
+  # @param user_id [Integer] ID of the user to be re-invited.
+  # @return [Token] The new token used for the invitation.
+  def reinvite_user(user_id)
+    token = Token.find_by user_id: user_id, action: token_action
+
+    token.destroy! if token
+    new_token = Token.create! user_id: user_id, action: token_action
```

Comment:
You are not handing those Exceptions for `destroy!` and `create!`. Is there a reason why you are preferring the bang methods over the non-bang? The bang methods are typically used only within tests or from the console and I don't see the benefit of using them here. 

BTW. You could write
```Token.where(user_id: user_id, action: token_action).destroy_all```

If you are not using the bang method.